src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -90,37 +97,36 @@ import scala.runtime.Statics
   *  // 5
   *  }}}
   *
-  *  There are a number of subtle points to the above example.
+  *  There are some subtleties here.
   *
-  *  - The definition of `fibs` is a `val` not a method.  The memoization of the
+  *  - The definition of `fibs` uses `val` not `def`.  The memoization of the
   *  `LazyList` requires us to have somewhere to store the information and a `val`
   *  allows us to do that.
   *
-  *  - While the `LazyList` is actually being modified during access, this does not
-  *  change the notion of its immutability.  Once the values are memoized they do
-  *  not change and values that have yet to be memoized still "exist", they
-  *  simply haven't been realized yet.
+  *  - Though the `LazyList` changes as it is accessed, this does not
+  *  contradict its immutability.  Once the values are memoized they do
+  *  not change. Values that have yet to be memoized still "exist", they
+  *  simply haven't been computed yet.
   *
-  *  - One must be cautious of memoization; you can very quickly eat up large
-  *  amounts of memory if you're not careful.  The reason for this is that the
-  *  memoization of the `LazyList` creates a structure much like
-  *  [[scala.collection.immutable.List]].  So long as something is holding on to
-  *  the head, the head holds on to the tail, and so it continues recursively.
-  *  If, on the other hand, there is nothing holding on to the head (e.g. we used
+  *  - One must be cautious of memoization; it can eat up memory if you're not
+  *  careful.  That's because memoization of the `LazyList` creates a structure much like
+  *  [[scala.collection.immutable.List]].  As long as something is holding on to
+  *  the head, the head holds on to the tail, and so on recursively.
+  *  If, on the other hand, there is nothing holding on to the head (e.g. if we used
   *  `def` to define the `LazyList`) then once it is no longer being used directly,
   *  it disappears.
   *
   *  - Note that some operations, including [[drop]], [[dropWhile]],
   *  [[flatMap]] or [[collect]] may process a large number of intermediate
   *  elements before returning.  These necessarily hold onto the head, since
-  *  they are methods on `LazyList`, and a lazy list holds its own head. For
-  *  computations of this sort where memoization is not desired, use
-  *  `Iterator` when possible.
+  *  they are methods on `LazyList`, and the list holds its own head. For
+  *  computations of this sort where memoization is not desired, consider
+  *  using `Iterator` instead.
```

Comment:
I'm not sure this is correct. I put in considerable effort to ensure that none of the methods mentioned here hold on to the head (see [here](https://github.com/scala/scala/blob/v2.13.2/src/library/scala/collection/immutable/LazyList.scala#L975-L1099)). to quote from the comment there:

> All of the following `<op>Impl` methods are carefully written so as not to leak the beginning of the `LazyList`. They copy the initial `LazyList` (`ll`) into `var rest`, which gets closed over as a `scala.runtime.ObjectRef`, thus not permanently leaking the head of the `LazyList`.

the methods do not actually process any elements before returning (due to being lazy), but rather have their non-head-retaining logic deferred until something forces the `LazyList`.

Reply:
For now I'm resolving this by simply omitting the following text:

```diff
-  * These necessarily hold onto the head, since                                           
-  *  they are methods on `LazyList`, and the list holds its own head. For                                              
-  *  computations of this sort where memoization is not desired, consider                                              
-  *  using `Iterator` instead.  
```

As it's clearly wrong as-is, but determining whether we still ought to say _something_ in this area isn't something I want to tackle today.